homeassistant/components/myuplink/__init__.py

```diff
@@ -34,12 +35,16 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
         )
     )
     session = config_entry_oauth2_flow.OAuth2Session(hass, config_entry, implementation)
+    auth = AsyncConfigEntryAuth(aiohttp_client.async_get_clientsession(hass), session)
+
+    try:
+        await auth.async_get_access_token()
+    except ClientError as err:
```

Comment:
We should preferably catch `ClientResponseError` first and raise `ConfigEntryAuthFailed` for status 401/403 as specified by the API. Raise `ConfigEntryNotReady` for other client errors.

Example:
https://github.com/home-assistant/core/blob/fd1f712d67adc16803a64c6275290b22466cb3f7/homeassistant/components/google/__init__.py#L168-L175